Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -3793,12 +3793,19 @@ Element const* Document::element_from_point(double x, double y)
 
     // 2. If there is a box in the viewport that would be a target for hit testing at coordinates x,y, when applying the transforms
     //    that apply to the descendants of the viewport, return the associated element and terminate these steps.
+    Optional<Painting::HitTestResult> hit_test_result;
     if (auto const* paintable_box = this->paintable_box(); paintable_box) {
-        if (auto result = paintable_box->hit_test(position, Painting::HitTestType::Exact); result.has_value()) {
-            if (auto* dom_node = result->dom_node(); dom_node && dom_node->is_element())
-                return static_cast<Element const*>(dom_node);
-        }
+        (void)paintable_box->hit_test(position, Painting::HitTestType::Exact, [&](Painting::HitTestResult result) {
```

Comment:
Is it cheap to copy a HitTestResult?

Reply:
it is a struct with pointer to `paintable`, `index_in_node` (int) and `internal_position` (enum), so it should be cheap to copy.